library/src/main/java/org/dbtools/android/domain/database/cursor/JdbcMemoryCursor.java

```diff
@@ -69,18 +75,20 @@ public int getPosition() {
 
     @Override
     public boolean move(int offset) {
-        currentPosition += offset;
-        return true;
+        return moveToPosition(currentPosition + offset);
     }
 
     @Override
     public boolean moveToPosition(int position) {
-        if (position >= 0 && position < getCount()) {
-            currentPosition = position;
-            return true;
+        int count = getCount();
+        if (position < 0) {
+            currentPosition = -1;
+        } else if (position >= count) {
+            currentPosition = count;
```

Comment:
Wouldn't this give you an index out of bounds??  (if I had an array of size 10... then if currentPosition got set to 10 (count).... then when you did a data.get(currentPosition) you would get an exception)